cohorts/survival.py

```diff
@@ -72,59 +77,89 @@ def plot_kmf(df,
         if is_median:
             label_suffix += " (median)"
         default_label_with_condition = "%s > %s" % (condition_col, label_suffix)
-    else:
+        with_condition_label = with_condition_label or default_label_with_condition
+        no_condition_label = no_condition_label or default_label_no_condition
+        label_map = {False: no_condition_label,
+                     True: with_condition_label}
+        color_map = {False: no_condition_color,
+                     True: with_condition_color}
+    elif df[condition_col].dtype == 'O' or df[condition_col].dtype.name == "category":
```

Comment:
Looks like a `lambda`-based solution would help cover the last two use cases in a similar way the other plotting utilities work; but if we don't immediately need it, I would say just leave that for another day/PR since the `threshold` argument might potentially complicate the implementation :man_shrugging: